### PR TITLE
Distinguish between `forc init` and `forc new` (Copy over changes from PR#1669)

### PR DESCRIPTION
**Pull request by ra0x3**
Up-to-date version of https://github.com/FuelLabs/sway/pull/1669

### Description
- This PR changes `forc init` to `forc new`, and updates `forc init` functionality to be more similar to `cargo init`

### Testing steps
- [ ] Try pulling down the PR and creating a `forc init` project with combinations of the different flags
----

Read the Docs:
- Sway Book: https://fuellabs.github.io/sway/latest
- Rust SDK Book: https://fuellabs.github.io/fuels-rs/latest
- TypeScript SDK: https://github.com/FuelLabs/fuels-ts

Join the Community:
- Follow us @SwayLang: https://twitter.com/SwayLang
- Ask questions in dev-chat on Discord: https://discord.com/invite/xfpK4Pe

Report Bugs:
- Sway Issues: https://github.com/FuelLabs/sway/issues/new add

### CODE REVIEW

**Review by mitchmindtree — CHANGES_REQUESTED** [review]
This appears to be missing the changes merged into #1669 via #1841.

Would you mind if we instead re-open #1669 and go from there? If your local copy is messed up, you can delete it and re-fetch from #1669.

In general, I think it's best if we try to keep the same work under a single PR as it's a little easier to keep track of what has been reviewed and what hasn't. If you're having git troubles, feel free to hmu and I'll see if I can assist!

**Comment by ra0x3** [comment]
> This appears to be missing the changes merged into #1669 via #1841.

@mitchmindtree I've cherry-picked #1841 over.

> In general, I think it's best if we try to keep the same work under a single PR as it's a little easier to keep track of what has been reviewed and what hasn't. If you're having git troubles, feel free to hmu and I'll see if I can assist!

I tend to agree. I think this work was unfortunately an example of something that was just sitting on the vine for too long.

**Review by mitchmindtree — APPROVED** [review]
> @mitchmindtree I've cherry-picked https://github.com/FuelLabs/sway/pull/1841 over.

Thanks for that!

Lgtm! There's one doc comment that didn't make it over, but not enough of an issue to block this any further :) will happily re-approve if you want to add it though.

File: forc/src/cli/commands/init.rs [hunk 1]
```diff
@@ -2,23 +2,29 @@ use crate::ops::forc_init;
 use anyhow::Result;
 use clap::Parser;
 
-/// Create a new Forc project.
+/// Create a new Forc project in an existing directory.
 #[derive(Debug, Parser)]
 pub struct Command {
+    #[clap(long)]
```
Comment (mitchmindtree) [inline forc/src/cli/commands/init.rs:8]:
```suggestion
    /// The directory in which the forc project will be initialized.
    #[clap(long)]
```

Comment (mitchmindtree) [inline forc/src/cli/commands/init.rs:8]:
```suggestion
    /// The directory in which the forc project will be initialized.
    #[clap(long)]
```

**Review by mitchmindtree — COMMENTED** [review]
Woops, looks like I forgot to add the actual line!

**Review by mitchmindtree — DISMISSED** [review]

**Review by mitchmindtree — APPROVED** [review]

**Review by JoshuaBatty — APPROVED** [review]

**Pull request merged**